### DISCUSSION TITLE
Data segmentation

### DISCUSSION THREAD

**Discussion by xiansenzhao** [Serial-Studio/Serial-Studio#161] [Q&A] [answered]
There are some ideas in use, would like to consult: Is it possible to divide a row of data according to keywords? This allows for more flexible use in existing projects. 
such as:
a123 b123 c123 d123 e123
For example, the above data is divided into multiple groups of data according to abcde.

**Comment by alex-spataru** [comment, marked as answer]
As of the latest version, you can achieve this by customizing the [Frame Parser](https://github.com/Serial-Studio/Serial-Studio/wiki/Project-Editor#frame-parser-function-view) and writing your own JavaScript code to categorize the frames according to your specific requirements. It took me some time to get it right, but I believe this approach offers users the flexibility they need for their projects, while still retaining the simplicity and capability to handle most frame formats or over-complicating the code.
